src/libstd/sync/once.rs

```diff
@@ -71,7 +71,7 @@ impl Once {
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn call_once<F>(&'static self, f: F) where F: FnOnce() {
         // Optimize common path: load is much cheaper than fetch_add.
-        if self.cnt.load(Ordering::SeqCst) < 0 {
+        if self.cnt.load(Ordering::Acquire) < 0 {
```

Comment:
Slightly more efficient would be to use a `Relaxed` load and only `fence(Acquire)` if the count is in fact negative.